mediapipe/graphs/instantmotiontracking/calculators/matrices_manager_calculator.cc

```diff
@@ -0,0 +1,324 @@
+    const DiagonalMatrix3f render_scaling_diagonal = GenerateScalingMatrix(GetDefaultRenderScale(render_data[render_idx++]));
+
+    // A vector representative of the translation of the object in OpenGL coordinate space must be generated
+    const Vector3f translation_vector = GenerateAnchorVector(anchor);
+
+    // Concatenate all model matrix data
+    const Matrix4fCM final_model_matrix = GenerateEigenModelMatrix(translation_vector, render_scaling_diagonal * user_transformed_rotation_submatrix);
+
+    // The generated model matrix must be mapped to TimedModelMatrixProto (col-wise)
+    for (int x = 0; x < final_model_matrix.rows(); ++x) {
+      for (int y = 0; y < final_model_matrix.cols(); ++y) {
+        model_matrix->add_matrix_entries(final_model_matrix(x, y));
+      }
+    }
+  }
+
+  cc->Outputs()
+      .Tag(kModelMatricesTag)
+      .Add(model_matrices.release(), cc->InputTimestamp());
+
+  return ::mediapipe::OkStatus();
+}
+
+// Using a specified rotation value in radians, generate a rotation matrix for use with base rotation submatrix
+const Matrix3f MatricesManagerCalculator::GenerateUserRotationMatrix(const float rotation_radians) {
+  Eigen::Matrix3f user_rotation_submatrix;
+    user_rotation_submatrix =
+        // The rotation in radians must be inverted to rotate the object
+        // with the direction of finger movement from the user (system dependent)
+        Eigen::AngleAxisf(-rotation_radians, Eigen::Vector3f::UnitY()) *
+        Eigen::AngleAxisf(0, Eigen::Vector3f::UnitZ()) *
+        Eigen::AngleAxisf(0, Eigen::Vector3f::UnitX());
+  // Matrix must be transposed due to the method of submatrix generation in Eigen
+  return user_rotation_submatrix.transpose();
+}
+
+// Using a specified scale factor, generate a scaling matrix for use with base rotation submatrix
+const DiagonalMatrix3f MatricesManagerCalculator::GenerateScalingMatrix(const float scale_factor) {
+  DiagonalMatrix3f scaling_matrix(scale_factor, scale_factor, scale_factor);
+  return scaling_matrix;
+}
+
+// TODO: Investigate possible differences in warping of tracking speed across screen
+// Using the sticker anchor data, a translation vector can be generated in OpenGL coordinate space
+const Vector3f MatricesManagerCalculator::GenerateAnchorVector(const Anchor tracked_anchor) {
+  // Using an initial_z value in OpenGL space, generate a new base z-axis value to mimic scaling by distance.
+  const float z = initial_z_ * tracked_anchor.z;
+
+  // Using triangle geometry, the minimum for a y-coordinate that will appear in the view field
+  // for the given z value above can be found.
+  const float y_minimum = z * (tan(vertical_fov_radians_ / 2));
+
+  // The aspect ratio of the device and y_minimum calculated above can be used to find the
+  // minimum value for x that will appear in the view field of the device screen.
+  const float x_minimum = y_minimum * (1.0 / aspect_ratio_);
```

Comment:
So aspect_ratio_ here is y / x? It's a bit different from our convention. Can you document this in the beginning of this file?